### ISSUE TITLE
🐛 Empty lines highlight plus-marker as whitespace-error

### ISSUE THREAD

**Issue by Nemo157** [dandavison/delta#388]
With a test file:

```
No Spaces

Spaces
  
```

(Note that there are spaces on the last line)

Running

```
delta --no-gitconfig --keep-plus-minus-markers <(echo -n) test-file
```

highlights the plus-marker on the empty line as a whitespace error

![temp](https://user-images.githubusercontent.com/81079/99059687-34c0a400-259f-11eb-9b2f-c6591a18302f.png)

**Comment by Kr1ss-XD** [comment]
I guess it *is* a whitespace error, since newlines are whitespace, no ?

**Comment by Nemo157** [comment]
It's very distracting having it highlighted in the middle of a block, and it's inconsistent with what `git` itself highlights as whitespace errors:

![image](https://user-images.githubusercontent.com/81079/99154062-3bd5d800-26ad-11eb-891e-22ab1148b44f.png)

![image](https://user-images.githubusercontent.com/81079/99154063-46906d00-26ad-11eb-9034-4e7628dc7674.png)

**Comment by Kr1ss-XD** [comment]
> It's very distracting having it highlighted in the middle of a block

Agreed, though one could also argue that it might be advantageous to have whitespace diffs highlighted esp. when working with languages as python where they are crucial

> it's inconsistent with what git itself highlights as whitespace errors

That depends on the configuration. Not sure though what the defaults are, to be honest, but IIRC git *does* highlight such a diff without any config.

**Comment by dandavison** [comment]
Hi @Nemo157, thanks for this!

@Kr1ss-XD what @Nemo157 is pointing out is that completely normal, not-at-all-an-error blank lines are being styled with an error background color when using `keep-plus-minus-markers`. I would say this is definitely a bug!

<table><tr><td><img width=400px src="https://user-images.githubusercontent.com/52205/99154396-745dbd80-267d-11eb-9134-987a1e9d38ab.png" alt="image" /></td></tr></table>

**Comment by Kr1ss-XD** [comment]
Oh, I see. I've missed the point then I guess.

**Comment by ericbn** [comment]
I think this is part of a bigger issue, as I reported in #345.

**Comment by dandavison** [comment]
@Nemo157 thanks for this, it is fixed in master (unreleased).

@ericbn thanks for pointing out the link to #345. The fix that has been merged for this branch fixes a subset of the bugs you demonstrated in #345, but not all.

**Comment by dandavison** [comment]
This fix is released now (v0.4.5).
